cmd/operator-sdk/new/cmd.go

```diff
@@ -548,3 +471,29 @@ func validateProject() error {
 	log.Info("Project validation successful.")
 	return nil
 }
+
+func verifyFlags() error {
+	if operatorType != projutil.OperatorTypeGo && operatorType != projutil.OperatorTypeAnsible && operatorType !=
+		projutil.OperatorTypeHelm {
+		return fmt.Errorf("value of --type can only be `go`, `ansible`, or `helm`: %v",
+			projutil.ErrUnknownOperatorType{Type: operatorType})
+	}
+	if operatorType != projutil.OperatorTypeAnsible && generatePlaybook {
+		return fmt.Errorf("value of --generate-playbook can only be used with --type `ansible`")
+	}
+
+	if err := fs.VerifyCommonFlags(operatorType); err != nil {
+		return err
+	}
+
+	if operatorType == projutil.OperatorTypeGo {
+		if len(fs.APIVersion) != 0 || len(fs.Kind) != 0 {
+			return fmt.Errorf("operators of type Go do not use --api-version or --kind")
+		}
+
+		if err := projutil.CheckRepo(repo); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
Nit: move this function back up above `func execProjCmd()` where it was before for easier `git diff/blame` in the future.